client/src/app/components/manage-organizations/organization-view/create-organization/create-organization.component.ts

```diff
@@ -26,20 +25,34 @@ export class CreateOrganizationComponent extends OrganizationViewComponent {
   }
 
   submitQuery(): void {
-    this.organizationService.createOrganization({
-      org_name: this.orgName,
-      address: this.location !== '' ? this.location : undefined,
-      status: true
-    }).subscribe(() => {
-      setTimeout(() => {
-        // Redirect user back to list of templates
-        this.router.navigate(['/manage-organizations']);
-      }, 1000);
-    }, err => {
-      if (err.error && err.error.org_name) {
-        this.orgError = 'An organization with this name already exists';
-      }
-    });
-  }
+    if (this.linesR.length > 0 && this.location === '') {
+      this.populateExcelElem();
+    } else if (this.linesR.length > 0 && this.location !== '') {
+      this.populateExcelElem();
+      this.populateUserInputElem();
+    } else if (this.linesR.length === 0 && this.location !== '') {
+      this.populateUserInputElem();
+    }
```

Comment:
I see these exact same checks being made twice in different places. You could extract these checks into their own function(s) so at least the function name(s) would convey information on their purpose (or you could simply add a comment if you want).

Reply:
Good catch! I will fix this.